airflow/providers/google/cloud/operators/dataproc.py

```diff
@@ -627,6 +681,11 @@ def execute(self, context):
             cluster = self._create_cluster(hook)
             self._handle_error_state(hook, cluster)
 
+        self.xcom_push(
+            context,
+            key="cluster_conf",
+            value={"cluster_name": self.cluster_name, "region": self.region, "project_id": self.project_id,},
+        )
```

Comment:
Let's move it to right after hook invocation. In this way, there will be cluster link no matter what is the state of the cluster. Only problem I see is when deleting cluster on error state but this is an edge case

Reply:
Similar flow with the jobs. Sure. Will update.

Reply:
Awesome, let me know once I can re-run DAG to check if the links work as expected

Reply:
@turbaszek Updated. Please take a look.